source/uwp/Renderer/lib/AdaptiveBackgroundImage.cpp

```diff
@@ -0,0 +1,97 @@
+#include "pch.h"
+#include "AdaptiveBackgroundImage.h"
+
+#include "Util.h"
+
+using namespace Microsoft::WRL;
+using namespace Microsoft::WRL::Wrappers;
+using namespace ABI::AdaptiveNamespace;
+using namespace ABI::Windows::Foundation;
+using namespace ABI::Windows::UI::Xaml;
+using namespace ABI::Windows::UI::Xaml::Controls;
+
+namespace AdaptiveNamespace
+{
+    AdaptiveBackgroundImage::AdaptiveBackgroundImage() {}
+
+    HRESULT AdaptiveBackgroundImage::RuntimeClassInitialize() noexcept try
+    {
+        std::shared_ptr<AdaptiveSharedNamespace::BackgroundImage> image =
+            std::make_shared<AdaptiveSharedNamespace::BackgroundImage>();
+        return RuntimeClassInitialize(image);
+    }
+    CATCH_RETURN;
+
+    _Use_decl_annotations_ HRESULT AdaptiveBackgroundImage::RuntimeClassInitialize(
+        const std::shared_ptr<AdaptiveSharedNamespace::BackgroundImage>& sharedImage) try
+    {
+        std::shared_ptr<AdaptiveSharedNamespace::BackgroundImage> image = sharedImage;
+        if (sharedImage == nullptr)
```

Comment:
>if (sharedImage == nullptr) [](start = 8, length = 27)

Did you hit this case? Looks like our other types just return E_INVALIDARG, and while this seems fine, i'm wondering if it should be consistent

Reply:
It doesn't work if I make it return an E_INVALIDARG. I think what makes this different is that normally objects are treated as AdaptiveCardElements and so it's weird if you say that you're going to use an Image but you set it to null. Here, every Card, Column, and Container has to have a BackgroundImage by default. In the event that the user doesn't explicitly say what the BackgroundImage is for the element (this case), it should just construct a "basic" background image object. Nothing will be rendered but all the properties are there in case you want to change them. Or should I change it to make it just outright return S_OK and the Card/Column/Container's backgroundImage property is just null?

---
In reply to: [253172646](https://github.com/Microsoft/AdaptiveCards/pull/2316#discussion_r253172646) [](ancestors = 253172646)

Reply:
I'd say if the background image is not set the property should be null. One way to handle this is follow the model of showcardactions which just doesn't try to make the subcard if it's null in the shared model. (AdaptiveShowCardAction::RuntimeClassInitialize)

---
In reply to: [254508406](https://github.com/Microsoft/AdaptiveCards/pull/2316#discussion_r254508406) [](ancestors = 254508406,253172646)